sdk/storage/azure-storage-common/src/storage_switch_to_secondary_policy.cpp

```diff
@@ -5,6 +5,8 @@
 
 namespace Azure { namespace Storage { namespace _internal {
 
+  Azure::Core::Context::Key const SecondaryHostReplicaStatusKey;
```

Comment:
Why do we define this again, isn't it already there in `storage_switch_to_secondary_policy.hpp` above?

Reply:
I don't think the Storage client library needs a key at all. We have a key in the retry policy and we have a static function which the storage policy can call to get the retry count. This internally uses a context key but Storage doesn't need to do anything in regards to this at all; just call the function that returns the retry count.

Reply:
This is how you declare static instances. Header has declaration, .cpp contains an actual instance.

Reply:
I'm saying that there is no need for a SecondaryHostReplicaStatusKey key at all.

Reply:
I'm sorry for the confusion Jeffrey. I was not arguing for or against - I was replying to Ahson's question.
In this PR, I am translating existing code to the new keys implementation.
Let us cover `SecondaryHostReplicaStatusKey` removal as a separate issue between our team (probably you and Victor, b/c of the recent changes to the Retry Context), and the Storage team.

Reply:
I opened an issue to track that - #2022